server/utils.go

```diff
@@ -174,3 +185,196 @@ func (w *withCode) Format(s fmt.State, verb rune) {
+					}
+					resourceAndVerb := map[string]map[string]oputil.DataAction{}
+					resourceAndVerb[resourceName] = map[string]oputil.DataAction{}
+					resourceAndVerb[resourceName][verb] = da
+					operationsMap[group] = append(operationsMap[group], resourceAndVerb)
+				}
+			}
+		}
+	}
+
+	klog.V(5).Infof("Operations list: %v", operationsMap)
+
+	return operationsMap, nil
+}
+
+func fetchApiResources() ([]*metav1.APIResourceList, error) {
+	// creates the in-cluster config
+	klog.V(5).Infof("Fetch apiresources list")
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubeconfig")
+	}
+
+	kubeclientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubernetes clientset")
+	}
+
+	apiresourcesList, err := kubeclientset.Discovery().ServerPreferredResources()
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(5).Infof("List of ApiResources fetched from apiserver: %v", apiresourcesList)
+
+	return apiresourcesList, nil
+}
+
+func fetchDataActionsList(environment string, clusterType string, loginURL string, tenantID string) ([]oputil.Operation, error) {
+	env := azure.PublicCloud
+	var err error
+	if environment != "" {
+		env, err = azure.EnvironmentFromName(environment)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse environment for Azure.")
+		}
+	}
+
+	endpoint := ""
+
+	switch clusterType {
+	case oputil.ConnectedClusters:
+		endpoint = fmt.Sprintf(oputil.OperationsEndpointFormatARC, env.ResourceManagerEndpoint)
+	case oputil.ManagedClusters:
+		endpoint = fmt.Sprintf(oputil.OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	case oputil.Fleets:
+		endpoint = fmt.Sprintf(oputil.OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	default:
+		return nil, errors.New("Failed to create endpoint for Get Operations call.")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create request for Get Operations call.")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", fmt.Sprintf("guard-%s-%s-%s", v.Version.Platform, v.Version.GoVersion, v.Version.Version))
+
+	var token string
+	if clusterType == oputil.ConnectedClusters {
+		tokenResp, err := oputil.GetAuthHeaderUsingMSIForARC(env.ResourceManagerEndpoint)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting authorization headers for Get Operations call.")
+		}
+
+		token = tokenResp.AccessToken
+	} else if clusterType == oputil.ManagedClusters || clusterType == oputil.Fleets {
+		tokenProvider := graph.NewAKSTokenProvider(loginURL, tenantID)
+
+		authResp, err := tokenProvider.Acquire("")
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting authorization headers for Get Operations call.")
+		}
+
+		token = authResp.Token
+	} else {
+		return nil, errors.New("Unsupported clusterType for Get Operations call.")
```

Comment:
this looks like a dead code as there is a switch earlier validating clusterType already